### ISSUE TITLE
Export  and re importing config on a fresh install causes issues

### ISSUE THREAD

**Issue by DemianTinkiel** [jenkinsci/configuration-as-code-plugin#1530]
### Your checklist for this issue

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Jenkins 2.249.3
- [x] 1.46

- [x] Fedora 30 / Docker jenkins/jenkins:lts

### Description

For context: I am reimporting existing config because I had the same issues with my custom configured instance, so as a test I tried with a fresh install (no changes to config) to check what I did wrong.

In a fresh install, with default config, exporting as yml and reimporting it shows `Found conflicting configuration at Yaml Source: /var/jenkins_home/jenkins.yml  in /var/jenkins_home/jenkins.yml` for several lines:
* primaryView
* excludeClientIPFromCrumb
* all of gitSCM
* all  of junitTestResultStorage
* adminAddress
* all of location
* all of mailer
* + 12 others

Even after fixing those "problems" (i.e removing those entries out of the YML) I get 
```
jenkins_1     | 2020-11-10 11:49:48.856+0000 [id=79]    WARNING o.e.j.s.h.ContextHandler$Context#log: Error while serving http://localhost:3030/configuration-as-code/checkNewSource
jenkins_1     | java.lang.NullPointerException
jenkins_1     |         at hudson.model.MyViewsProperty.save(MyViewsProperty.java:138)
jenkins_1     |         at hudson.BulkChange.commit(BulkChange.java:98)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:271)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.configure(DataBoundConfigurator.java:83)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$doConfigure$16668e2$1(HeteroDescribableConfigurator.java:277)
jenkins_1     |         at io.vavr.CheckedFunction0.lambda$unchecked$52349c75$1(CheckedFunction0.java:247)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.doConfigure(HeteroDescribableConfigurator.java:277)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$null$2(HeteroDescribableConfigurator.java:86)
jenkins_1     |         at io.vavr.control.Option.map(Option.java:392)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$configure$3(HeteroDescribableConfigurator.java:86)
jenkins_1     |         at io.vavr.Tuple2.apply(Tuple2.java:238)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.configure(HeteroDescribableConfigurator.java:83)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.configure(HeteroDescribableConfigurator.java:55)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:345)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:276)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.configure(DataBoundConfigurator.java:83)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:345)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:276)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.configure(DataBoundConfigurator.java:83)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$doConfigure$16668e2$1(HeteroDescribableConfigurator.java:277)
jenkins_1     |         at io.vavr.CheckedFunction0.lambda$unchecked$52349c75$1(CheckedFunction0.java:247)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.doConfigure(HeteroDescribableConfigurator.java:277)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$null$2(HeteroDescribableConfigurator.java:86)
jenkins_1     |         at io.vavr.control.Option.map(Option.java:392)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$configure$3(HeteroDescribableConfigurator.java:86)
jenkins_1     |         at io.vavr.Tuple2.apply(Tuple2.java:238)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.configure(HeteroDescribableConfigurator.java:83)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.check(HeteroDescribableConfigurator.java:92)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.check(HeteroDescribableConfigurator.java:55)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:351)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.check(BaseConfigurator.java:287)
jenkins_1     |         at io.jenkins.plugins.casc.ConfigurationAsCode.lambda$checkWith$8(ConfigurationAsCode.java:755)
jenkins_1     |         at io.jenkins.plugins.casc.ConfigurationAsCode.invokeWith(ConfigurationAsCode.java:691)
jenkins_1     |         at io.jenkins.plugins.casc.ConfigurationAsCode.checkWith(ConfigurationAsCode.java:755)
jenkins_1     |         at io.jenkins.plugins.casc.ConfigurationAsCode.checkWith(ConfigurationAsCode.java:629)
jenkins_1     |         at io.jenkins.plugins.casc.ConfigurationAsCode.doCheckNewSource(ConfigurationAsCode.java:237)
jenkins_1     |         at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:627)
jenkins_1     |         at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:396)
jenkins_1     | Caused: java.lang.reflect.InvocationTargetException
jenkins_1     |         at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:400)
jenkins_1     |         at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:408)
jenkins_1     |         at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:212)
jenkins_1     |         at org.kohsuke.stapler.SelectionInterceptedFunction$Adapter.invoke(SelectionInterceptedFunction.java:36)
jenkins_1     |         at org.kohsuke.stapler.verb.HttpVerbInterceptor.invoke(HttpVerbInterceptor.java:48)
jenkins_1     |         at org.kohsuke.stapler.SelectionInterceptedFunction.bindAndInvoke(SelectionInterceptedFunction.java:26)
jenkins_1     |         at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:145)
jenkins_1     |         at org.kohsuke.stapler.MetaClass$11.doDispatch(MetaClass.java:536)
jenkins_1     |         at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:58)
jenkins_1     |         at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:766)
jenkins_1     |         at org.kohsuke.stapler.Stapler.invoke(Stapler.java:898)
jenkins_1     |         at org.kohsuke.stapler.MetaClass$9.dispatch(MetaClass.java:457)
jenkins_1     |         at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:766)
jenkins_1     |         at org.kohsuke.stapler.Stapler.invoke(Stapler.java:898)
jenkins_1     |         at org.kohsuke.stapler.Stapler.invoke(Stapler.java:694)
jenkins_1     |         at org.kohsuke.stapler.Stapler.service(Stapler.java:240)
jenkins_1     |         at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
jenkins_1     |         at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:763)
jenkins_1     |         at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1631)
jenkins_1     |         at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:154)
jenkins_1     |         at org.jenkinsci.plugins.ssegateway.Endpoint$SSEListenChannelFilter.doFilter(Endpoint.java:248)
jenkins_1     |         at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
jenkins_1     |         at jenkins.security.ResourceDomainFilter.doFilter(ResourceDomainFilter.java:76)
jenkins_1     |         at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
jenkins_1     |         at jenkins.telemetry.impl.UserLanguages$AcceptLanguageFilter.doFilter(UserLanguages.java:129)
jenkins_1     |         at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
jenkins_1     |         at io.jenkins.blueocean.ResourceCacheControl.doFilter(ResourceCacheControl.java:134)
jenkins_1     |         at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
jenkins_1     |         at io.jenkins.blueocean.auth.jwt.impl.JwtAuthenticationFilter.doFilter(JwtAuthenticationFilter.java:60)
jenkins_1     |         at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:151)
jenkins_1     |         at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:157)
jenkins_1     |         at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
jenkins_1     |         at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:153)
jenkins_1     |         at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
jenkins_1     |         at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
jenkins_1     |         at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)
jenkins_1     |         at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
jenkins_1     |         at jenkins.security.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:119)
jenkins_1     |         at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
jenkins_1     |         at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)
jenkins_1     |         at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
jenkins_1     |         at org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMeProcessingFilter.java:142)
jenkins_1     |         at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
jenkins_1     |         at org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.java:271)
jenkins_1     |         at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
jenkins_1     |         at jenkins.security.BasicHeaderProcessor.doFilter(BasicHeaderProcessor.java:93)
jenkins_1     |         at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
jenkins_1     |         at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)
jenkins_1     |         at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:67)
jenkins_1     |         at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
jenkins_1     |         at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:90)
jenkins_1     |         at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:171)
jenkins_1     |         at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
jenkins_1     |         at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:51)
jenkins_1     |         at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
jenkins_1     |         at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:82)
jenkins_1     |         at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
jenkins_1     |         at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)
jenkins_1     |         at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
jenkins_1     |         at jenkins.security.SuspiciousRequestFilter.doFilter(SuspiciousRequestFilter.java:36)
jenkins_1     |         at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
jenkins_1     |         at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:549)
jenkins_1     |         at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
jenkins_1     |         at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:578)
jenkins_1     |         at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
jenkins_1     |         at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
jenkins_1     |         at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1610)
jenkins_1     |         at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
jenkins_1     |         at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1369)
jenkins_1     |         at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
jenkins_1     |         at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:489)
jenkins_1     |         at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1580)
jenkins_1     |         at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
jenkins_1     |         at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1284)
jenkins_1     |         at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
jenkins_1     |         at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
jenkins_1     |         at org.eclipse.jetty.server.Server.handle(Server.java:501)
jenkins_1     |         at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
jenkins_1     |         at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:556)
jenkins_1     |         at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
jenkins_1     |         at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:272)
jenkins_1     |         at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
jenkins_1     |         at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
jenkins_1     |         at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
jenkins_1     |         at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
jenkins_1     |         at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
jenkins_1     |         at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
jenkins_1     |         at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
jenkins_1     |         at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
jenkins_1     |         at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
jenkins_1     |         at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
jenkins_1     |         at java.lang.Thread.run(Thread.java:748)
jenkins_1     | 2020-11-10 11:49:48.861+0000 [id=79]    WARNING h.i.i.InstallUncaughtExceptionHandler#handleException: Caught unhandled exception with ID 3bb21d5e-30be-4805-8280-edf5ff49a9e4
jenkins_1     | java.lang.NullPointerException
jenkins_1     |         at hudson.model.MyViewsProperty.save(MyViewsProperty.java:138)
jenkins_1     |         at hudson.BulkChange.commit(BulkChange.java:98)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:271)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.configure(DataBoundConfigurator.java:83)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$doConfigure$16668e2$1(HeteroDescribableConfigurator.java:277)
jenkins_1     |         at io.vavr.CheckedFunction0.lambda$unchecked$52349c75$1(CheckedFunction0.java:247)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.doConfigure(HeteroDescribableConfigurator.java:277)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$null$2(HeteroDescribableConfigurator.java:86)
jenkins_1     |         at io.vavr.control.Option.map(Option.java:392)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$configure$3(HeteroDescribableConfigurator.java:86)
jenkins_1     |         at io.vavr.Tuple2.apply(Tuple2.java:238)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.configure(HeteroDescribableConfigurator.java:83)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.configure(HeteroDescribableConfigurator.java:55)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:345)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:276)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.configure(DataBoundConfigurator.java:83)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:345)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:276)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.configure(DataBoundConfigurator.java:83)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$doConfigure$16668e2$1(HeteroDescribableConfigurator.java:277)
jenkins_1     |         at io.vavr.CheckedFunction0.lambda$unchecked$52349c75$1(CheckedFunction0.java:247)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.doConfigure(HeteroDescribableConfigurator.java:277)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$null$2(HeteroDescribableConfigurator.java:86)
jenkins_1     |         at io.vavr.control.Option.map(Option.java:392)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$configure$3(HeteroDescribableConfigurator.java:86)
jenkins_1     |         at io.vavr.Tuple2.apply(Tuple2.java:238)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.configure(HeteroDescribableConfigurator.java:83)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.check(HeteroDescribableConfigurator.java:92)
jenkins_1     |         at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.check(HeteroDescribableConfigurator.java:55)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:351)
jenkins_1     |         at io.jenkins.plugins.casc.BaseConfigurator.check(BaseConfigurator.java:287)
jenkins_1     |         at io.jenkins.plugins.casc.ConfigurationAsCode.lambda$checkWith$8(ConfigurationAsCode.java:755)
jenkins_1     |         at io.jenkins.plugins.casc.ConfigurationAsCode.invokeWith(ConfigurationAsCode.java:691)
jenkins_1     |         at io.jenkins.plugins.casc.ConfigurationAsCode.checkWith(ConfigurationAsCode.java:755)
jenkins_1     |         at io.jenkins.plugins.casc.ConfigurationAsCode.checkWith(ConfigurationAsCode.java:629)
jenkins_1     |         at io.jenkins.plugins.casc.ConfigurationAsCode.doCheckNewSource(ConfigurationAsCode.java:237)
```

I understand it might not be directlly usable jenkins.yml for things like API tokens but if we can't export basic config like just booleans, what is the point? Am I making a wrong assumption on how to use the plugin?

**Comment by jetersen** [comment]
an export is not 100% copy paste, I suggest taking a single section/plugin at a time and see what causes the issue, once you found the issue. Would be nice if you could share.

Looks like you have an issue with importing views or in particular user views properties.

Without knowing what jenkins.yaml your trying to import I am little help to you.

**Comment by timja** [comment]
Some of the views properties aren't compatible yet.

That stacktrace looks like `myView` property

**Comment by alon-dotan-starkware** [comment]
any news regarding this one? how to bypass it?

**Comment by timja** [comment]
only include what you need, not everything.

**Comment by brandon-fryslie** [comment]
It says

> Export is not intended to offer a directly usable jenkins.yaml configuration. It can be used for inspiration writing your own, be aware export can be partial, or fail for some components.

On the JCasC page in Jenkins.  This can probably be closed

**Comment by basil** [comment]
Fixed in https://github.com/jenkinsci/jenkins/pull/6202. Released in 2.344.
